### ISSUE TITLE
Memory leaks when running CLI on sitemap

### ISSUE THREAD

**Issue by gagan0123** [GoogleChromeLabs/ps-analysis-tool#258]
**Describe the bug**
When running the CLI tool for a sitemap, there are memory leaks errors being displayed in the CLI output

**To Reproduce**
1. `npm run cli -- -s https://gagan.pro/sitemaps/rakuten.xml`.

**Expected behavior**
There shouldn't be any errors or warnings in the CLI output

**Screenshots**
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/1535505/03bac49d-5089-4b57-8060-4af8a83e0b4c)

**Desktop (please complete the following information):**
 - OS: Pop!_OS 22.04 LTS
 - Browser: Google Chrome
 - Version: 117.0.5938.149

<details>
  <summary><b>Entire Log</b></summary>
  

  ```
/home/gagan/.nvm/versions/node/v18.17.0/bin/npm run cli -- -s https://gagan.pro/sitemaps/rakuten.xml

> ps-analysis-tool@0.3.2 cli
> node dist/cli/index.js -s https://gagan.pro/sitemaps/rakuten.xml

✓ Done parsing Sitemap
Provided sitemap has 39 pages. Please enter the number of pages you want to analyze (Default 39): 
✓ Done analysing cookies in urls 1 - 3 
✓ Done analysing cookies in urls 4 - 6 
✓ Done analysing cookies in urls 7 - 9 
✓ Done analysing cookies in urls 10 - 12 
✓ Done analysing cookies in urls 13 - 15 
✓ Done analysing cookies in urls 16 - 18 
✓ Done analysing cookies in urls 19 - 21 
✓ Done analysing cookies in urls 22 - 24 
✓ Done analysing cookies in urls 25 - 27 
✓ Done analysing cookies in urls 28 - 30 
✓ Done analysing cookies in urls 31 - 33 
✓ Done analysing cookies in urls 34 - 36 
✓ Done analysing cookies in urls 37 - 39 
✓ Done technology in urls 1 - 3 
✓ Done technology in urls 4 - 6 
✓ Done technology in urls 7 - 9 
✓ Done technology in urls 10 - 12 
(node:262410) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 16 SIGINT listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:262410) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 16 SIGTERM listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(node:262410) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 16 SIGHUP listeners added to [process]. Use emitter.setMaxListeners() to increase limit
✓ Done technology in urls 13 - 15 
(node:262410) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 16 exit listeners added to [process]. Use emitter.setMaxListeners() to increase limit
✓ Done technology in urls 16 - 18 
✓ Done technology in urls 19 - 21 
✓ Done technology in urls 22 - 24 
✓ Done technology in urls 25 - 27 
✓ Done technology in urls 28 - 30 
✓ Done technology in urls 31 - 33 
✓ Done technology in urls 34 - 36 
✓ Done technology in urls 37 - 39 
✓ Done analysing technologies
Report is being served at the URL: http://localhost:9000?dir=rakuten-co.jp&type=sitemap
  ```
</details>

**Comment by ayushnirwal** [comment]
The fix for this bug has been implemented and merged to `develop` branch and is ready for release.

**Comment by maitreyie-chavan** [comment]
Released as a part of [v0.4.0](https://github.com/GoogleChromeLabs/ps-analysis-tool/releases/tag/v0.4.0), closing this issue.
